### ISSUE TITLE
Container linter fails to lint the kernel service

### ISSUE THREAD

**Issue by derrabus** [symfony/symfony#40868]
**Symfony version(s) affected**: 5.3-dev

**Description**
Running the container linter on a new symfony app fails with the following error:

> Invalid definition for service "kernel": "Symfony\Component\HttpKernel\Kernel::__construct()" requires 2 arguments, 0 passed.

**How to reproduce**

```sh
symfony new --version=next container-lint-reproducer
cd container-lint-reproducer
./bin/console lint:container
```

**Comment by maxim-dovydenok** [comment]
Since 5.3 App\Kernel Definition is instance of ChildDefinition (parent = .instanceof.App\Kernel.0.kernel)
Because of that, it is processed by `ResolveChildDefinitionsPass`. And this CompilerPass replaces the kernel definition
```php
        $def = new Definition();

        // merge in parent definition
        // purposely ignored attributes: abstract, shared, tags, autoconfigured
        $def->setClass($parentDef->getClass());
        $def->setArguments($parentDef->getArguments());
        $def->setMethodCalls($parentDef->getMethodCalls());
        $def->setProperties($parentDef->getProperties());
        if ($parentDef->isDeprecated()) {
            $deprecation = $parentDef->getDeprecation('%service_id%');
            $def->setDeprecated($deprecation['package'], $deprecation['version'], $deprecation['message']);
        }
        $def->setFactory($parentDef->getFactory());
        $def->setConfigurator($parentDef->getConfigurator());
        $def->setFile($parentDef->getFile());
        $def->setPublic($parentDef->isPublic());
        $def->setLazy($parentDef->isLazy());
        $def->setAutowired($parentDef->isAutowired());
        $def->setChanges($parentDef->getChanges());
...
```
Notice that `synthetic` property is not copied. This issue can be fixed by adding `$def->setSynthetic($parentDef->isSynthetic());`

Is it a correct solution?

**Comment by stof** [comment]
synthetic services should be excluded from the autoconfiguration entirely and so not using a ChildDefinition, as they are not configured by the container anyway. The right fix is to stop making it a ChildDefinition.

**Comment by nicolas-grekas** [comment]
Fixed in #41177
